Components/Overlay/src/OgreOverlayElement.cpp

```diff
@@ -329,10 +331,19 @@ namespace Ogre {
             return;
 
         mMaterial->load();
+
+        auto dstPass = mMaterial->getBestTechnique()->getPass(0);
+        if (dstPass->getLightingEnabled() || dstPass->getDepthCheckEnabled())
+        {
+            LogManager::getSingleton().logWarning(
+                "force-disabling 'lighting' and 'depth_check' of Material " + mat->getName() +
+                " for use with OverlayElement " + getName());
+        }
+
         // Set some prerequisites to be sure
-        mMaterial->setLightingEnabled(false);
+        dstPass->setLightingEnabled(false);
         mMaterial->setReceiveShadows(false);
-        mMaterial->setDepthCheckEnabled(false);
+        dstPass->setDepthCheckEnabled(false);
```

Comment:
Isn't it changed logic now?
The calls to the material changes all techniques with all passes.
Now you're just changing one technique with one pass.
Or do I get something wrong?

Reply:
the idea was to make this specific to the actually used pass. But the pass might change if the user switches the viewport scheme. So I should probably leave this alone.